src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -653,7 +653,6 @@ private static MethodSymbol SubstituteTypeParameters(MethodSymbol method)
 
             private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
```

Comment:
I find the name of this method very confusing. It looks like its logic is specific to Anonymous Types, perhaps the name should reflect this and asserting on the input night be appropriate.